src/ReinforcementLearningCore/src/policies/q_based_policies/learners/approximators/neural_network_approximator.jl

```diff
@@ -59,4 +60,63 @@ RLBase.update!(app::ActorCritic, gs) = Flux.Optimise.update!(app.optimizer, para
 function Base.copyto!(dest::ActorCritic, src::ActorCritic)
     Flux.loadparams!(dest.actor, params(src.actor))
     Flux.loadparams!(dest.critic, params(src.critic))
+end
+
+#####
+# GaussianNetwork
+#####
+
+"""
+    GaussianNetwork(;pre=identity, μ, logσ)
+
+Returns `μ` and `logσ` when called. 
+Create a distribution to sample from 
+using `Normal.(μ, exp.(logσ))`.
+"""
+Base.@kwdef struct GaussianNetwork{P,U,S}
+    pre::P = identity
+    μ::U
+    logσ::S
+end
+
+Flux.@functor GaussianNetwork
+
+function (m::GaussianNetwork)(S)
+    x = m.pre(S)
+    m.μ(x), m.logσ(x) 
+end
+
+"""
+This function is compatible with a multidimensional action space.
+"""
+function evaluate(model::GaussianNetwork, state)
```

Comment:
The function name `evaluate` seems to be too general to me. Maybe we can simply use functional call here?

- Also, I think we can add a keyword argument `is_return_log_prob=true` to control whether to return `logp_π` or not. I think in some cases, we only need the actions.
- Doc string should also be updated to make it clear that the `tanh` is applied.
- A positional argument `rng::AbstractRNG` should also be added to ensure reproducibility.

Reply:
Well. I will complete it.

Reply:
What does `functional call` means? @findmyway

Reply:
In addition, do I need to update the version of RLCore?

Reply:
```
(model::GaussianNetwork)(state; is_return_log_prob=true) = ...
(model::GaussianNetwork)(rng, state; kw...) = model(GLOBAL_RNG, state;kw...)
```

Reply:
Doesn't this conflict with the following code?
```julia
function (m::GaussianNetwork)(S)
    x = m.pre(S)
    m.μ(x), m.logσ(x) 
end
```

Reply:
Hmm, you are right. Can these two functions be merged somehow?Or just give it a more specific name?

Reply:
Maybe we can use this form:
```julia
(model::GaussianNetwork)(state, is_return_log_prob=true; rng) = ...
```